File: c/problems/problem#763.c
#include <stdlib.h>

int *partitionLabels(char *s, int *returnSize) {
  int lastIdx[26];
  for (int i = 0; s[i]; i++) {
    int c = s[i] - 97;
    lastIdx[c] = i;
  }
  int sz = 0;
  int last = 0;
  for (int i = 0; s[i]; i++) {
    int c = s[i] - 97;
    if (lastIdx[c] > last) {
      last = lastIdx[c];
    }
    if (last == i) {
      sz++;
    }
  }
  int *ans = malloc(sz * sizeof(int));
  int j = 0;
  last = 0;
  int count = 0;
  for (int i = 0; s[i]; i++) {
    count++;
    int c = s[i] - 97;
    if (lastIdx[c] > last) {
      last = lastIdx[c];
    }
    if (last == i) {
      ans[j] = count;
      j++;
      count = 0;
    }
  }
  *returnSize = sz;
  return ans;
}

File: go/problem#763.go
package main

func partitionLabels(s string) []int {
	lastExist := [26]int{}
	for i, c := range s {
		lastExist[int(c)-97] = i
	}
	last := 0
	count := 0
	ans := []int{}
	for i, c := range s {
		count++
		le := lastExist[int(c)-97]
		if le > last {
			last = le
		}
		if i == last {
			ans = append(ans, count)
			count = 0
		}
	}
	return ans
}
